exercises/leap/tests/leap.rs

```diff
@@ -37,6 +37,11 @@ fn test_year_divisible_by_400_leap_year() {
 #[ignore]
 fn test_any_old_year() {
     assert_eq!(leap::is_leap_year(1997), false);
+    assert_eq!(leap::is_leap_year(1), false);
+    assert_eq!(leap::is_leap_year(4), true);
```

Comment:
> The algorithm is not necessarily valid for years in the Julian calendar, such as years before 1752 in the British Empire. The year 1700 was a leap year in the Julian calendar, but not in the Gregorian calendar.

In the spirit of the exercise, given Gregorian calendar specification in the Readme, perhaps introducing a specification that prior to 1582 it should fail.  As in ordinary use, we don't go back before that date, and when going back, we have to make adjustments.  Let's keep the exercise simple, it is about logic, more than dates.  Calendaring opens up a lot of nuances, and I believe the Gregorian calendar was used with this in mind, and as a limitation of expectation.

And since the exercise is not about date display, I don't think it makes sense to apply ISO 8601 since that is a display standard.  We are reporting true or false, only, right?

> The reason why I raised the issue is not due to the Gregorian calendar mention, but more the programming issues with the assignment. The assignment uses an integer for implementation which allows 0 and values to the maximum available size, so a programmer would have to handle the values and corner cases no matter the interpretation of the Gregorian calendar.

Then let's limit the integer use, maybe.  A test that holds the upper bound the Gregorian calendar is meant to be used for.  And a lower bound that starts at adoption.

Otherwise we really do need to introduce everything that has been hinted at in this discussion, and even more comes to mind, on a global level, how do we get the corresponding date for the other common calendars of the world?  Do they even have the idea of a leap year?  Since this exercise isn't "find out the date", but "is it a leap year?" I think we can restrict the use inside the Gregorian calandar, and since the idea is to not use the built in Date facilities, we don't need to worry about the different display formats.

On the other hand, the subject is extremely interesting, and I might enjoy exploring date conversions, finding out of a date was a leap year in year -4.  But it should have been actually a leap year in history, not what would have been a leap year based on Gregorian rules.

Reply:
I have put the tests into a separate tests. I have left out the test of year zero.